modules/phase_field/test/tests/KKS_system/kks_barrier_height_material.i

```diff
@@ -0,0 +1,243 @@
+#
+# KKS toy problem with barrier height defined as a material
+#
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 15
+  ny = 15
+  nz = 0
+  xmin = -2.5
+  xmax = 2.5
+  ymin = -2.5
+  ymax = 2.5
+  zmin = 0
+  zmax = 0
+  elem_type = QUAD4
+[]
+
+[AuxVariables]
+  [./Fglobal]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[Variables]
+  # order parameter
+  [./eta]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+
+  # hydrogen concentration
+  [./c]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+
+  # chemical potential
+  [./w]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+
+  # hydrogen phase concentration (matrix)
+  [./cm]
+    order = FIRST
+    family = LAGRANGE
+    initial_condition = 0.0
+  [../]
+  # hydrogen phase concentration (delta phase)
+  [./cd]
+    order = FIRST
+    family = LAGRANGE
+    initial_condition = 0.0
+  [../]
+[]
+
+[ICs]
+  [./eta]
+    variable = eta
+    type = SmoothCircleIC
+    x1 = 0.0
+    y1 = 0.0
+    radius = 1.5
+    invalue = 0.2
+    outvalue = 0.1
+    int_width = 0.75
+  [../]
+  [./c]
+    variable = c
+    type = SmoothCircleIC
+    x1 = 0.0
+    y1 = 0.0
+    radius = 1.5
+    invalue = 0.6
+    outvalue = 0.4
+    int_width = 0.75
+  [../]
+[]
+
+
+[BCs]
+  [./Periodic]
+    [./all]
+      variable = 'eta w c cm cd'
+      auto_direction = 'x y'
+    [../]
+  [../]
+[]
+
+[Materials]
+  # Free energy of the matrix
+  [./fm]
+    type = DerivativeParsedMaterial
+    f_name = fm
+    args = 'cm'
+    function = '(0.1-cm)^2'
+  [../]
+
+  # Free energy of the delta phase
+  [./fd]
+    type = DerivativeParsedMaterial
+    f_name = fd
+    args = 'cd'
+    function = '(0.9-cd)^2'
+  [../]
+
+  # h(eta)
+  [./h_eta]
+    type = SwitchingFunctionMaterial
+    h_order = HIGH
+    eta = eta
+  [../]
+
+  # g(eta)
+  [./g_eta]
+    type = BarrierFunctionMaterial
+    g_order = SIMPLE
+    eta = eta
+  [../]
+
+  # constant properties
+  [./constants]
+    type = GenericConstantMaterial
+    prop_names  = 'M   L   kappa barrier_height'
+    prop_values = '0.7 0.7 0.4   0.4'
```

Comment:
This is not a good test. We need to test the _new_ aspect of this capability, which is a spatially varying barrier height.

Reply:
Makes sense, though I don't have a use case for that yet. The test would just be some arbitrary variation.

Reply:
OK, added a test with a barrier height that depends on concentration.